cmake_test/template_expectfail.txt

```diff
@@ -0,0 +1,37 @@
+#This file is a template intended to be configured via CMakeTest.
```

Comment:
Maybe make `template` a subdirectory of `cmake_test` and shorten the name of the template files?

Reply:
Should be fixed now. I've also temporarily disabled the tests that have yet to be converted so the test checks are once again accurate, I did not realize I had accidentally enabled them. I did find a bug in one of the asserts though which is causing one of the tests to fail, I will be fixing that later this week. Other than that one assert it should be ready for use. The code itself is a little long because I prioritized getting it functional quickly over beauty, but I'll be splitting it up into more files this week as well.

Reply:
That all sounds good!!

Reply:
Bug has been fixed, turns out it was a problem with the way I was executing the sections in the subprocess but I've changed it to where everything works as expected now. I have also split up some of the subprocess configuration code to make it easier to manage.